Source/monster.cpp

```diff
@@ -1359,16 +1343,27 @@ bool MonsterAttack(int monsterId)
 	auto &monster = Monsters[monsterId];
 
 	if (monster.animInfo.currentFrame == monster.data().animFrameNum - 1) {
-		MonsterAttackPlayer(monsterId, monster.enemy, monster.toHit, monster.minDamage, monster.maxDamage);
+		if ((monster.flags & MFLAG_TARGETS_MONSTER) != 0)
+			MonsterAttackMonster(monster, Monsters[monster.enemy], monster.toHit, monster.minDamage, monster.maxDamage);
+		else
+			MonsterAttackPlayer(monster, Players[monster.enemy], monster.toHit, monster.minDamage, monster.maxDamage);
 		if (monster.ai != AI_SNAKE)
 			PlayEffect(monster, 0);
 	}
 	if (IsAnyOf(monster.type().type, MT_NMAGMA, MT_YMAGMA, MT_BMAGMA, MT_WMAGMA) && monster.animInfo.currentFrame == 8) {
-		MonsterAttackPlayer(monsterId, monster.enemy, monster.toHit + 10, monster.minDamage - 2, monster.maxDamage - 2);
+		if ((monster.flags & MFLAG_TARGETS_MONSTER) != 0)
+			MonsterAttackMonster(monster, Monsters[monster.enemy], monster.toHit + 10, monster.minDamage - 2, monster.maxDamage - 2);
+		else
+			MonsterAttackPlayer(monster, Players[monster.enemy], monster.toHit + 10, monster.minDamage - 2, monster.maxDamage - 2);
+
 		PlayEffect(monster, 0);
 	}
 	if (IsAnyOf(monster.type().type, MT_STORM, MT_RSTORM, MT_STORML, MT_MAEL) && monster.animInfo.currentFrame == 12) {

```

Comment:
Do these monsters really get a double strike or is there a way to clean this up to avoid the duplication?

At the very least this should be possible:
```cpp
void MonsterAttackEnamy(Monster &monster, int toHit, int minDamage, int maxDamage)
{
	if ((monster.flags & MFLAG_TARGETS_MONSTER) != 0)
		MonsterAttackMonster(monster, Monsters[monster.enemy], toHit, minDamage, maxDamage);
	else
		MonsterAttackPlayer(monster, Players[monster.enemy], toHit, minDamage, maxDamage);
}
```

Reply:
hmm, what do you mean by "double strike"?

Reply:
mAFNum for the monsters listed here are 2, 2, 2, 4, 8, 8, 12, 12. So if they are also hardcoded to execute `MonsterAttackPlayer` on 9/13 that means that they will land two hits per animation.

Reply:
I added `MonsterAttackEnemy`

Reply:
I am not sure about the double strike, but the code should be behaving the same. I just moved a check that was done inside `MonsterAttackPlayer` to the caller